### PR TITLE
dmd.target: Move all top-level imports to local selective imports

### PR DESCRIPTION
**Pull request by ibuclaw**
@WalterBright

### CODE REVIEW

File: src/dmd/target.d [hunk 1]
```diff
@@ -399,22 +387,24 @@ extern (C++) struct Target
      */
     extern (C++) uint alignsize(Type type)
     {
+        import dmd.mtype : TY = ENUMTY;
```
Comment (RazvanN7) [inline src/dmd/target.d:390]: This import appears 5 times in this PR. I don't know what your threshold is, but it looks like a good candidate for the struct level imports.

Comment (RazvanN7) [reply src/dmd/target.d:390]: Also, if we go down this road, does this mean we can get rid of all the T* aliases in dmd.mtype ?

File: src/dmd/target.d [hunk 2]
```diff
@@ -783,26 +785,25 @@ extern (C++) struct Target
...
+            if (tns.ty == TY.Tcomplex32)
                 return true;
             if (tns.isscalar())
                 return false;
 
             tns = tns.baseElemOf();
-            if (tns.ty == Tstruct)
+            if (auto ts = tns.isTypeStruct())
             {
-                StructDeclaration sd = (cast(TypeStruct)tns).sym;
```
Comment (RazvanN7) [inline src/dmd/target.d:800]: The diff would have been smaller if this line became: `StructDeclaration sd = ts.sym`

Comment (ibuclaw) [reply src/dmd/target.d:800]: `StructDeclaration` needs importing though.

Comment (RazvanN7) [reply src/dmd/target.d:800]: `auto` ?

File: src/dmd/target.d (hunk 1 above)
Comment (ibuclaw) [reply src/dmd/target.d:390]: Someone tried removing `ENUMTY` before, can give it a retry. #12579

File: src/dmd/target.d (hunk 2 above)
Comment (ibuclaw) [reply src/dmd/target.d:800]: I don't see it worth it for a two line saving.

**Comment by ibuclaw** [comment]
@andralex since you said you liked the idea of this on the ML. ;-)

**Review by RazvanN7 — APPROVED** [review]
Looks good to me, but I'll defer the merger to @WalterBright or @andralex

**Comment by WalterBright** [comment]
I'm not that big a fan of moving the imports into local contexts, but it's ok with me.

**Comment by andralex** [comment]
noice

**Pull request merged**